tests/annotator/annotator.cpp

```diff
@@ -1911,42 +1984,21 @@ TEST(Annotator, hashChangesAndUpdates)
     auto model = parser->parseModel(modelStringUniqueIds);
     auto annotator = libcellml::Annotator::create();
 
-    EXPECT_FALSE(annotator->isBuilt());
-
-    annotator->buildModelIndex(model);
-    EXPECT_TRUE(annotator->isBuilt());
+    EXPECT_FALSE(annotator->hasModel());
 
-    model->setId("iHaveChangedSinceTheLastBuild");
-    EXPECT_FALSE(annotator->isBuilt());
+    // Model is set and the id map is not created.
+    annotator->setModel(model);
 
-    // Can set id automatically, but this does not trigger a rebuild.
-    EXPECT_EQ("b4da55", annotator->assignId(libcellml::Annotator::Type::COMPONENT, model->component(0)));
-    EXPECT_FALSE(annotator->isBuilt());
+    // Can set id manually, this does not trigger a rebuild of the id map.
+    model->setId("iHaveChanged");
 
-    // Shouldn't be able to retrieve anything since the build is out of date.
+    // The id map is built when a query is made.
     EXPECT_EQ(nullptr, annotator->model("model_1"));
-    EXPECT_EQ(nullptr, annotator->encapsulation("encapsulation_1"));
-    EXPECT_EQ(nullptr, annotator->component("component_1"));
-    EXPECT_EQ(nullptr, annotator->component("component_2"));
-    EXPECT_EQ(nullptr, annotator->component("component_ref_1"));
-    EXPECT_EQ(nullptr, annotator->component("component_3"));
-    EXPECT_EQ(nullptr, annotator->componentRef("component_ref_2"));
-    EXPECT_EQ(nullptr, annotator->importSource("import_1"));
-    EXPECT_EQ(nullptr, annotator->units("units_1"));
-    EXPECT_EQ(nullptr, annotator->importSource("import_2"));
-    EXPECT_EQ(nullptr, annotator->units("units_2"));
-    EXPECT_EQ(nullptr, annotator->unit("unit_1").first);
-    EXPECT_EQ(size_t(0), annotator->unit("unit_1").second);
-    EXPECT_EQ(nullptr, annotator->variable("variable_1"));
-    EXPECT_EQ(nullptr, annotator->variable("variable_2"));
-    EXPECT_EQ(nullptr, annotator->reset("reset_1"));
-    EXPECT_EQ(nullptr, annotator->testValue("test_value_1"));
-    EXPECT_EQ(nullptr, annotator->resetValue("reset_value_1"));
-    EXPECT_EQ(nullptr, annotator->variable("variable_3"));
-    EXPECT_EQ(nullptr, annotator->variable("variable_4"));
-    EXPECT_EQ(nullptr, annotator->connection("connection_1").first);
-    EXPECT_EQ(nullptr, annotator->mapVariables("map_variables_1").first);
-    EXPECT_EQ(nullptr, annotator->mapVariables("map_variables_2").first);
+
+    // Can set id automatically, this does not trigger a rebuild of the id map.
+    EXPECT_EQ("b4da55", annotator->assignId(model->component(0)));
+
+    EXPECT_NE(nullptr, annotator->encapsulation("encapsulation_1"));
```

Comment:
I don't understand this?  The old id should still be present and able to be retrieved since it's only the model id which was changed outside of the annotator?